src/getdata.nim

```diff
@@ -24,42 +24,38 @@
 # IN THE SOFTWARE.
 #
 
-import httpclient, asyncdispatch, json, times, strformat, strutils, options
+import httpclient, json, times, strformat, strutils, options
 
-proc returnWebSiteData(webUrl: string, Wthr: var WeatherObj): string =
+import types, utils
+
+proc returnWebSiteData*(webUrl: string, w: Weather): string =
   ##
   ## PROCEDURE: returnWebSiteData
   ## Input: final URL for DarkSky site to obtain forecast and WeatherObject
   ## Returns: raw web page body recieved and daily API calls count
   ## Description: open the provided URL returning the web site content received
   ## also extract HTTP header to obtain API calls total. API calls total is
-  ## updated directly into the WeatherObj (Wthr) passed to the proc.
+  ## updated directly into the Weather (w) passed to the proc.
   ##
-  var client = newAsyncHttpClient()
-  let response = waitfor client.get(webUrl)
-  let rawWebData = waitfor response.body
-  let webHeaders = response.headers
+  var client = newHttpClient()
```

Comment:
You don't need to use async if your app is not async itself - HttpClient can be used in synchronous code too